tests/test_rpc.c

```diff
@@ -65,20 +131,25 @@ test_lock(void **state)
     assert_int_equal(msgtype, NC_MSG_REPLY);
     assert_null(op);
     assert_string_equal(LYD_NAME(lyd_child(envp)), "ok");
+
     lyd_free_tree(envp);
 
-    /* send request on session #2 */
-    msgtype = nc_send_rpc(nc_sess2, rpc, 1000, &msgid);
+    /* request to lock from another session should fail when lock already */
+    msgtype = nc_send_rpc(st->nc_sess2, rpc, 1000, &msgid);
     assert_int_equal(msgtype, NC_MSG_RPC);
 
-    /* receive reply */
-    msgtype = nc_recv_reply(nc_sess2, rpc, msgid, 2000, &envp, &op);
+    /* recieve reply, should yield error */
+    msgtype = nc_recv_reply(st->nc_sess2, rpc, msgid, 2000, &envp, &op);
     assert_int_equal(msgtype, NC_MSG_REPLY);
     assert_null(op);
+    assert_string_equal(LYD_NAME(lyd_child(envp)), "rpc-error");
     assert_int_equal(LY_SUCCESS, lyd_print_mem(&str, envp, LYD_XML, LYD_PRINT_SHRINK));
-    lyd_free_tree(envp);
+
+    char expected[500];
+    sprintf(expected, LOCK_FAIL_TEMPLATE, msgid, nc_session_get_id(st->nc_sess));
```

Comment:
Unsafe and no reason for it, no problem with allocating the string dynamically (a few lines below). Also, variable declarations can only be at block start (in this case function start) but you can reuse `str2` or rename it to `expected`.